Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Ammo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Ammo.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Ammo.java	(revision )
@@ -1,6 +1,9 @@
 
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.AmmoModel;
+import net.deludobellico.commandops.estabeditor.model.Element;
+
 import javax.xml.bind.annotation.XmlAccessType;
 import javax.xml.bind.annotation.XmlAccessorType;
 import javax.xml.bind.annotation.XmlAttribute;
@@ -38,7 +41,7 @@
     "minOrderQty",
     "minOrderWeight"
 })
-public class Ammo {
+public class Ammo implements Element<AmmoModel> {
 
     @XmlElement(required = true)
     protected String name;
@@ -147,4 +150,8 @@
         this.id = value;
     }
 
+    @Override
+    public AmmoModel getModel() {
+        return new AmmoModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Force.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Force.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Force.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.ForceModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -78,7 +81,7 @@
         "ammoList",
         "canBombard"
 })
-public class Force {
+public class Force implements Element<ForceModel>{
 
     @XmlElement(required = true)
     protected String name;
@@ -645,4 +648,8 @@
         return this.flags;
     }
 
+    @Override
+    public ForceModel getModel() {
+        return new ForceModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/FormationEffects.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/FormationEffects.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/FormationEffects.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.FormationEffectsModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -39,7 +42,7 @@
         "targetPercentages",
         "security"
 })
-public class FormationEffects {
+public class FormationEffects implements Element<FormationEffectsModel>{
 
     @XmlElement(name = "moving-cohesion-level")
     protected double movingCohesionLevel;
@@ -223,4 +226,8 @@
         return this.flags;
     }
 
+    @Override
+    public FormationEffectsModel getModel() {
+        return new FormationEffectsModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Image.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Image.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Image.java	(revision )
@@ -1,6 +1,9 @@
 
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.ImageModel;
+
 import javax.xml.bind.annotation.XmlAccessType;
 import javax.xml.bind.annotation.XmlAccessorType;
 import javax.xml.bind.annotation.XmlAttribute;
@@ -27,7 +30,7 @@
  */
 @XmlAccessorType(XmlAccessType.FIELD)
 @XmlType(name = "Image")
-public class Image {
+public class Image implements Element<ImageModel> {
 
     @XmlAttribute(name = "id", required = true)
     protected int id;
@@ -74,4 +77,8 @@
         this.fileId = value;
     }
 
+    @Override
+    public ImageModel getModel() {
+        return new ImageModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Side.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Side.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Side.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.SideModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -42,7 +45,7 @@
         "defaultEnemyAarmFp",
         "nation"
 })
-public class Side {
+public class Side implements Element<SideModel>{
 
     @XmlElement(required = true)
     protected String name;
@@ -254,4 +257,8 @@
         return this.flags;
     }
 
+    @Override
+    public SideModel getModel() {
+        return new SideModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Vehicle.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Vehicle.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Vehicle.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.VehicleModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -76,7 +79,7 @@
         "battleWeight",
         "armour"
 })
-public class Vehicle {
+public class Vehicle implements Element<VehicleModel> {
 
     @XmlElement(required = true)
     protected String name;
@@ -594,4 +597,8 @@
         return this.flags;
     }
 
+    @Override
+    public VehicleModel getModel() {
+        return new VehicleModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Weapon.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Weapon.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Weapon.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.WeaponModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -56,7 +59,7 @@
         "mustDeployToFire",
         "performanceList"
 })
-public class Weapon {
+public class Weapon implements Element<WeaponModel> {
 
     @XmlElement(required = true)
     protected String name;
@@ -411,4 +414,8 @@
         return this.flags;
     }
 
+    @Override
+    public WeaponModel getModel() {
+        return new WeaponModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Nation.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Nation.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Nation.java	(revision )
@@ -1,6 +1,9 @@

 package net.deludobellico.commandops.estabeditor.data.jaxb;

+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.NationModel;
+
 import java.util.ArrayList;
 import java.util.List;
 import javax.xml.bind.annotation.XmlAccessType;
@@ -45,7 +48,7 @@
     "smallInsignia",
     "service"
 })
-public class Nation {
+public class Nation implements Element<NationModel> {

     @XmlElement(required = true)
     protected String name;
@@ -257,4 +260,8 @@
         return this.flags;
     }

+    @Override
+    public NationModel getModel() {
+        return new NationModel(this);
+    }
 }
Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Service.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Service.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Service.java	(revision )
@@ -1,6 +1,9 @@

 package net.deludobellico.commandops.estabeditor.data.jaxb;

+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.ServiceModel;
+
 import java.util.ArrayList;
 import java.util.List;
 import javax.xml.bind.annotation.XmlAccessType;
@@ -47,7 +50,7 @@
     "defaultIconColors",
     "force"
 })
-public class Service {
+public class Service implements Element<ServiceModel> {

     @XmlElement(required = true)
     protected String name;
@@ -285,4 +288,8 @@
         return this.flags;
     }

+    @Override
+    public ServiceModel getModel() {
+        return new ServiceModel(this);
+    }
 }
