Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Force.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Force.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Force.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.ForceModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -78,7 +81,7 @@
         "ammoList",
         "canBombard"
 })
-public class Force {
+public class Force implements Element<ForceModel>{
 
     @XmlElement(required = true)
     protected String name;
@@ -645,4 +648,8 @@
         return this.flags;
     }
 
+    @Override
+    public ForceModel getModel() {
+        return new ForceModel(this);
+    }
 }
