Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Service.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Service.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Service.java	(revision )
@@ -1,6 +1,9 @@
 
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.ServiceModel;
+
 import java.util.ArrayList;
 import java.util.List;
 import javax.xml.bind.annotation.XmlAccessType;
@@ -47,7 +50,7 @@
     "defaultIconColors",
     "force"
 })
-public class Service {
+public class Service implements Element<ServiceModel> {
 
     @XmlElement(required = true)
     protected String name;
@@ -285,4 +288,8 @@
         return this.flags;
     }
 
+    @Override
+    public ServiceModel getModel() {
+        return new ServiceModel(this);
+    }
 }
