Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Side.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Side.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Side.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.SideModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -42,7 +45,7 @@
         "defaultEnemyAarmFp",
         "nation"
 })
-public class Side {
+public class Side implements Element<SideModel>{
 
     @XmlElement(required = true)
     protected String name;
@@ -254,4 +257,8 @@
         return this.flags;
     }
 
+    @Override
+    public SideModel getModel() {
+        return new SideModel(this);
+    }
 }
