Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Vehicle.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Vehicle.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Vehicle.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.VehicleModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -76,7 +79,7 @@
         "battleWeight",
         "armour"
 })
-public class Vehicle {
+public class Vehicle implements Element<VehicleModel> {
 
     @XmlElement(required = true)
     protected String name;
@@ -594,4 +597,8 @@
         return this.flags;
     }
 
+    @Override
+    public VehicleModel getModel() {
+        return new VehicleModel(this);
+    }
 }
