Index: src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Weapon.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Weapon.java	(revision )
+++ src/main/java/net/deludobellico/commandops/estabeditor/data/jaxb/Weapon.java	(revision )
@@ -1,5 +1,8 @@
 package net.deludobellico.commandops.estabeditor.data.jaxb;
 
+import net.deludobellico.commandops.estabeditor.model.Element;
+import net.deludobellico.commandops.estabeditor.model.WeaponModel;
+
 import javax.xml.bind.annotation.*;
 import java.util.ArrayList;
 import java.util.List;
@@ -56,7 +59,7 @@
         "mustDeployToFire",
         "performanceList"
 })
-public class Weapon {
+public class Weapon implements Element<WeaponModel> {
 
     @XmlElement(required = true)
     protected String name;
@@ -411,4 +414,8 @@
         return this.flags;
     }
 
+    @Override
+    public WeaponModel getModel() {
+        return new WeaponModel(this);
+    }
 }
